Mathlib/Algebra/Category/GroupCat/Biproducts.lean

```diff
@@ -76,6 +76,9 @@ noncomputable def biprodIsoProd (G H : AddCommGroupCat.{u}) :
   IsLimit.conePointUniqueUpToIso (BinaryBiproduct.isLimit G H) (binaryProductLimitCone G H).isLimit
 #align AddCommGroup.biprod_iso_prod AddCommGroupCat.biprodIsoProd
 
+-- simpNF now complains after leanprover/lean4#2644
+attribute [nolint simpNF] AddCommGroupCat.biprodIsoProd_hom_apply
```

Comment:
This is genuinely a bad simp lemma; this should link the mathlib4 issue too, as I did in the suggested comment.

Reply:
Maybe we should actually fix this by writing the simps lemmas by hand?

Reply:
What is the threshold for "poorly typed"?

Reply:
Have a look at the type of the lemma (either before or after the bump) and see https://github.com/leanprover-community/mathlib4/issues/7657; the problem is when a morphism ` f : A -> B` ends up with a simp lemma of the form `ohno (a : NotAUnlessYouSquint) : f a = b`.

Reply:
Sure. But that seems like half of mathlib.

Reply:
Can you point to a lemma in mathlib with that problem that isn't touched by this PR?

Reply:
Ah ok. `transparency := reducible` vs `transparency := default` or above